### PR TITLE
GP-3740 dockerize and add tests

### PR DESCRIPTION
**Pull request by grapigeau**
Part of guanyin maintenance epic

### CODE REVIEW

File: test/setup_test_db.sh [hunk 1]
```diff
@@ -4,7 +4,17 @@
...
+
+# stop and remove any previously-running test containers
+docker stop "${PG_DB_CONTAINER_NAME}"
+docker rm "${PG_DB_CONTAINER_NAME}"
+
+# set up the Postgres database
+docker run --name "${PG_DB_CONTAINER_NAME}" -e POSTGRES_USER=test -e POSTGRES_PASSWORD=test -e POSTGRES_DB=test_report_db -d -p 5436:5432 postgres:10
+
+#perform cleaning and database migration
+docker run --rm -v $(pwd)/test:/flyway/conf -v $(pwd)/sql:/flyway/sql --network=host flyway/flyway clean && \
```
Comment (callunity) [inline test/setup_test_db.sh:17]: I found that the version of Docker Flyway that I'm running needed the `-cleanDisabled=false` argument in order to run `clean`. If your version doesn't require it but permits it as an option, would you mind adding that line to this script?

**Review by callunity — COMMENTED** [review]
The test additions and name/purpose declaration clarifications all look great!

File: test/setup_test_db.sh [hunk 2]
```diff
@@ -4,7 +4,17 @@
...
-docker run --rm -v $(pwd)/test:/flyway/conf -v $(pwd)/sql:/flyway/sql --network=host boxfuse/flyway clean && \
-docker run --rm -v $(pwd)/test:/flyway/conf -v $(pwd)/test/sql:/flyway/sql --network=host boxfuse/flyway migrate
+PG_DB_CONTAINER_NAME="test"  # Also used in package.json to stop the container after the test
+
+# stop and remove any previously-running test containers
+docker stop "${PG_DB_CONTAINER_NAME}"
+docker rm "${PG_DB_CONTAINER_NAME}"
+
+# set up the Postgres database
+docker run --name "${PG_DB_CONTAINER_NAME}" -e POSTGRES_USER=test -e POSTGRES_PASSWORD=test -e POSTGRES_DB=test_report_db -d -p 5436:5432 postgres:10
```
[the remark this answers is not in the document]
Comment (grapigeau) [reply test/setup_test_db.sh]: done, should be matching nabu now as well

File: test/setup_test_db.sh [hunk 3]
```diff
@@ -1,10 +1,30 @@
...
+docker stop "${PG_DB_CONTAINER_NAME}" || true  # it's fine if the container isn't running
+
+# create flyway.conf file
+echo "flyway.url=jdbc:postgresql://${DB_HOST}:${DB_PORT}/${DB_NAME}" > test/flyway.conf
+echo "flyway.user=${DB_USER}" >> test/flyway.conf
+echo "flyway.password=${DB_PW}" >> test/flyway.conf
+echo "flyway.cleanDisabled=false" >> test/flyway.conf
+
+# stop and remove any previously-running test containers
+docker stop "${PG_DB_CONTAINER_NAME}"
```
Comment (callunity) [inline test/setup_test_db.sh:20]: You've got a similar command up on line 11, maybe one of them could be removed?

**Review by callunity — APPROVED** [review]

**Comment by callunity** [comment]
A very minor request: would you mind adding a `npm posttest` script [(here's an example)](https://github.com/oicr-gsi/nabu/blob/f6b028bba916065fc997c508d316f1b19f5ff516/test/stop_test_dbs.sh) to clean up the container after tests complete, so that the port is freed up after tests complete? (Otherwise I have to manually kill the Guanyin test container before I run Nabu tests)

**Review by mlaszloffy — APPROVED** [review]

**Pull request merged**